hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/kafka/KafkaConsumerRecordToMessageConverter.java

```diff
@@ -17,30 +17,33 @@ public class KafkaConsumerRecordToMessageConverter {
     private volatile Subscription subscription;
     private final Map<String, KafkaTopic> topics;
     private final MessageContentReader messageContentReader;
+    private final KafkaHeaderExtractor kafkaHeaderExtractor;
     private final Clock clock;
 
     public KafkaConsumerRecordToMessageConverter(Topic topic,
                                                  Subscription subscription,
                                                  Map<String, KafkaTopic> topics,
                                                  MessageContentReader messageContentReader,
+                                                 KafkaHeaderExtractor kafkaHeaderExtractor,
                                                  Clock clock) {
         this.topic = topic;
         this.subscription = subscription;
         this.topics = topics;
         this.messageContentReader = messageContentReader;
+        this.kafkaHeaderExtractor = kafkaHeaderExtractor;
         this.clock = clock;
     }
 
     public Message convertToMessage(ConsumerRecord<byte[], byte[]> record, long partitionAssignmentTerm) {
         KafkaTopic kafkaTopic = topics.get(record.topic());
         UnwrappedMessageContent unwrappedContent = messageContentReader.read(record, kafkaTopic.contentType());
         return new Message(
-                unwrappedContent.getMessageMetadata().getId(),
+                kafkaHeaderExtractor.extractMessageId(record.headers()),
                 topic.getQualifiedName(),
                 unwrappedContent.getContent(),
                 kafkaTopic.contentType(),
                 unwrappedContent.getSchema(),
-                unwrappedContent.getMessageMetadata().getTimestamp(),
+                kafkaHeaderExtractor.extractTimestamp(record.headers()),
```

Comment:
Why not to use `ConsumerRecord.timestamp` instead of dedicated header? Or at least be possible to use it (pass whole ConsumerRecord to this method)

Reply:
Hey Arek, good point, I wasn't aware of this field, thanks! :)

On the frontend side we still have to set `ts` (timestamp) header for now as other our system reads it. However in the near future we will be able to remove the header as well.